### DISCUSSION TITLE
MySQL UUID binary encoding incompatible with string parser?

### DISCUSSION THREAD

**Discussion by rynoV** [rusuly/MySqlCdc#37] [Q&A]
Hi, thanks for making this project!

I'm trying to use this library with `UUID` values stored in `BINARY(16)` columns in MySql 8.0.35. The bytes are being parsed [here](https://github.com/rusuly/MySqlCdc/blob/46cb108d75256410a4f0187fdef8a038202fffbd/src/MySqlCdc/Protocol/PacketReader.cs#L302) as UTF8, but I believe it's failing because [UUID uses utf8mb3](https://dev.mysql.com/doc/refman/8.4/en/miscellaneous-functions.html#function_uuid), which is different from `System.Text.Encoding.UTF8`. For example, a UUID may include bytes like `0x9c` and `0x9d`, both of which get parsed to `0xfffd`:

```f#
let b: byte[] = [| byte 0x9c; byte 0x9d |]

let s = Encoding.UTF8.GetString(b)

for ch in s do
    printfn $"%08x{int16 ch} "

>> 0000fffd 
>> 0000fffd 
```

Have I misunderstood something?

I'm currently planning to make a fork that uses only byte arrays instead of `String` for string-typed columns

**Comment by rynoV** [comment]
Here's the fork: https://github.com/rusuly/MySqlCdc/compare/main...Auspice-Capital:MySqlCdc:main

**Comment by rusuly** [comment]
Hi, thanks for your contribution! I merged your fork and uploaded libs to the nuget. Sorry for very late response and slow support, there are reasons for that
